src/PrestaShopBundle/Component/CsvResponse.php

```diff
@@ -0,0 +1,259 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Component;
+
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+class CsvResponse extends StreamedResponse
+{
+    // Mode used to paginate page per page, 1/100, 2/100, 3/000, etc
+    const MODE_PAGINATION = 1;
+
+    // Mode used to paginate by offset, 1/100, 100/100, 200/100, etc (like MySql limit)
+    const MODE_OFFSET = 2;
+
+    /**
+     * @var array() CSV content
+     */
+    private $data;
+
+    /**
+     * @var String Export filename
+     */
+    private $fileName;
+
+    /**
+     * @var array
+     */
+    private $headersData = array();
+
+    /**
+     * @var int, self::MODE_PAGINATION by default
+     */
+    private $modeType = self::MODE_PAGINATION;
+
+    /**
+     * @var int, default 1 because default modeType is pagination
+     */
+    private $start = 1;
```

Comment:
Could be interesting to set a non-sense value to know that it needs to be replaced with the proper default value from $modeType